### PR TITLE
Generate: v4.38 removals and related updates

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

(see title)

### CODE REVIEW

File: src/transformers/models/opt/modeling_opt.py [hunk 1]
```diff
@@ -129,8 +129,8 @@ def _handle_deprecated_argument(config_arg_name, config, fn_arg_name, kwargs):
             val = None
             if fn_arg_name in kwargs:
                 logging.warning(
-                    "Passing in {} to {self.__class__.__name__} is deprecated and won't be supported from v4.38."
-                    " Please set it in the config instead"
+                    "Passing in {fn_arg_name} to {self.__class__.__name__} is deprecated and won't be supported from "
+                    "v4.39. Please set it in the config instead"
```
Comment (gante) [inline src/transformers/models/opt/modeling_opt.py:133]: This warning was not being print correctly (missing `fn_arg_name`), so I've pushed the deprecation cycle by 1 minor version

File: src/transformers/generation/__init__.py [hunk 2]
```diff
@@ -40,6 +40,11 @@
         "BeamSearchScorer",
         "ConstrainedBeamSearchScorer",
     ]
+    _import_structure["candidate_generator"] = [
+        "AssistedCandidateGenerator",
+        "CandidateGenerator",
+        "PromptLookupCandidateGenerator",
+    ]
```
Comment (gante) [inline src/transformers/generation/__init__.py:47]: These are now importable as `from transformers.generation import ...` (not as `from transformers import ...`). This is needed for tests (see below), but I don't think they are worth a global import.

File: src/transformers/generation/candidate_generator.py [hunk 3]
```diff
@@ -99,7 +99,8 @@ def __init__(
         # Make sure all data at the same device as assistant model
         device = assistant_model.device
         input_ids = input_ids.to(device)
-        inputs_tensor = inputs_tensor.to(device)
+        if inputs_tensor is not None:
+            inputs_tensor = inputs_tensor.to(device)
```
Comment (gante) [inline src/transformers/generation/candidate_generator.py:103]: `inputs_tensor` is an optional input

File: src/transformers/generation/utils.py [hunk 4]
```diff
@@ -4300,7 +4300,6 @@ def constrained_beam_search(
     def assisted_decoding(
         self,
         input_ids: torch.LongTensor,
-        assistant_model: Optional["PreTrainedModel"] = None,
```
Comment (gante) [inline src/transformers/generation/utils.py:4303]: removed according to the deprecation cycle -> remove all references and docstrings

File: src/transformers/utils/import_utils.py [hunk 5]
```diff
@@ -665,14 +665,6 @@ def is_flash_attn_greater_or_equal_2_10():
     return version.parse(importlib.metadata.version("flash_attn")) >= version.parse("2.1.0")
 
 
-def is_flash_attn_available():
-    logger.warning(
-        "Using `is_flash_attn_available` is deprecated and will be removed in v4.38. "
-        "Please use `is_flash_attn_2_available` instead."
-    )
-    return is_flash_attn_2_available()
```
Comment (gante) [inline src/transformers/utils/import_utils.py:673]: there were no further references in the code of this one

**Review by ArthurZucker — APPROVED** [review]
🧼 nice!

**Pull request merged**